services/auths.js

```diff
@@ -0,0 +1,40 @@
+const db = require("../data/dbConfig");
+const bcrypt = require("bcryptjs");
+const sendConfirmationEmail = require("../routes/utils/verificationEmail");
+const { verifyUser } = require("../data/models/user-model");
+
+exports.registerUser = async (req, res) => {
+  let { email, password, username } = req.body;
+
+  try {
+    const hash = bcrypt.hashSync(password, 10);
+    const [id] = await db("users").insert(
+      {
+        email,
+        username,
+        password: hash
+      },
+      "id"
+    );
+    const [user] = await db("users").where({ id });
+    sendConfirmationEmail(email, id);
+    return res.status(201).json(user);
+  } catch (error) {
+    res.status(500).json({
+      error: error.message
+    });
+  }
+};
+
+exports.verifyEmail = async (req, res) => {
+  const { id } = req.params;
```

Comment:
The service methods should have nothing to do with requests and response, leave that to the route or network layer, refer back to this article https://dev.to/santypk4/bulletproof-node-js-project-architecture-4epf#service